File: solutions/0389. Find the Difference/0389_1.cpp
// Time complexity: O(n)
// Space complexity: O(1)
class Solution {
 public:
  char findTheDifference(string_view s, string_view t) {
    const char sXors = accumulate(s.begin(), s.end(), 0, bit_xor<>());
    const char tXors = accumulate(t.begin(), t.end(), 0, bit_xor<>());
    return sXors ^ tXors;
  }
};

File: solutions/0389. Find the Difference/0389_1.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
  def findTheDifference(self, s: str, t: str) -> str:
    sXors = chr(functools.reduce(lambda x, y: x ^ y, map(ord, s), 0))
    tXors = chr(functools.reduce(lambda x, y: x ^ y, map(ord, t), 0))

    return chr(ord(sXors) ^ ord(tXors))  